rs/hang/src/import/manifest/mod.rs

```diff
@@ -0,0 +1,41 @@
+use reqwest::Client;
+use url::Url;
+
+use crate::{BroadcastProducer, Result};
+
+pub mod hls;
+
+pub enum Protocol {
+	Hls(hls::Hls),
+}
+
+/// A generic interface for importing a manifest (playlist) into a hang broadcast.
+pub struct ImportManifest {
+	protocol: Protocol,
+}
+
+impl ImportManifest {
+	pub fn new(broadcast: BroadcastProducer, format: &str, url: Url, client: Client) -> Option<Self> {
+		let protocol = match format {
+			"hls" => {
```

Comment:
IMO too generic, just make this HLS specific so it's more type safe. We can make a separate DASH module later.